### DISCUSSION TITLE
OpenFileDialog with 0.10.19 - Directory Property

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#11779] [Q&A] [answered]
Hi,

still using 0.10.19!

Writing the following code

```
        var dialog = new OpenFileDialog
        {
            Directory = "/point/to/inital/folder/"
        };
```

I would expect that when the dialog is opened the content of the folder given to property Directory (/point/to/inital/folder/) is shown.

But that's not the case. The content of its parent folder (/point/to/inital/) is shown. 

To get what I expect I have to add a pseudo folder to the path, i.e. "/point/to/inital/folder/pseudo"

Bug or Feature?

Joachim

**Comment by timunie** [comment]
Which OS? On windows, I've never seen it. However, if it is a bug, we cannot fix it for 0.10.x anymore as our process on getting 11.0 finished is far ahead now.

**Comment by drei97** [comment]
it is on KUbuntu 23.04.

Since I have already a workaround it is not necessary (for me) to fix it. But please take care that the bug is not taken also to 11.0

**Reply by timunie** [reply]
Can you try 11.0 RC in a simple sample to confirm it is solved there? I think so, but I can't test on your system.

**Reply by drei97** [reply]
Can you give me please a short hint, how to switch from 0.10.xx to 11.0? 

In NuGet I can download the libraries and the templates are on GitHub, but it seems that I am too blind to find any hints how to set up and to create a new 11.0 project.

**Reply by drei97** [reply]
sry, I got it now!

**Comment by drei97** [comment]
1) OpenFileDialog in 11.0 under KUbuntu 23.04 ingnores Directory property and opens $HOME/Documents. If Directory is pointing to a non existing path, the App crashes.

2) I am getting "15:25:28 Building /home/joachim/nas/08_source/01_Ausprobieren/dotNET/FileOpenApp/FileOpenApp.csproj
0>Von "obj/Debug/net7.0/refint/FileOpenApp.dll" konnte keine MVID extrahiert werden. Sind Sie sicher, dass es sich dabei um eine Referenzassembly handelt?
0>FileOpenApp -> /home/joachim/nas/08_source/01_Ausprobieren/dotNET/FileOpenApp/bin/Debug/net7.0/FileOpenApp.dll"

means, from my application dll there could not be extraced a MVID. The app is starting anyway.

3) There is warning that OpenFileDialog is outdated , I get "CS0618: Class 'Avalonia.Controls.OpenFileDialog' is obsolete: Use Window.StorageProvider API or TopLevel.StorageProvider API"

**Reply by timunie** [reply]
Exactly. You need to use TopLevel.StorageProvider now in 11.0. Refer to migration guide and breaking changes for detials. 
https://github.com/AvaloniaUI/Avalonia/wiki/Avalonia-11-Porting-Guide
https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes 

I have also a draft sample in here: https://github.com/timunie/Avalonia.Samples/tree/feature/MVVM_Dialogs/src/Avalonia.Samples/MVVM/MvvmDialogSample 

Let me know your thoughts. 

Viele Grüße 
Tim

**Comment by drei97** [comment]
I am playing around now

```
private async void Button_OnClick(object? sender, RoutedEventArgs e)
{        
    var storageProvider =  TopLevel.GetTopLevel(this)!.StorageProvider;
    //IStorageFolder? options = await StorageProvider.TryGetFolderFromPathAsync("/home/joachim/nag/");
    //IStorageFolder? options = await storageProvider.TryGetFolderFromPathAsync("/home/joachim/nas/");
    IStorageFolder? options = await storageProvider.TryGetWellKnownFolderAsync(WellKnownFolder.Downloads);
    Debug.Print(options?.Path.AbsolutePath);
        
    var d = StorageProvider.OpenFilePickerAsync(new()
    {
        SuggestedStartLocation = options
    });
}
```

My resumee so far:

1) Build process takes very very long for such a short code
2) given path is not opened. The dialog still opens $HOME/Documents in every case
3) well known folder Downloads not found

hmm

**Reply by timunie** [reply]
@maxkatz6 do you think this is worth to open an issue for?

**Reply by antondie** [reply]
I have seen this also in Linux Mint, but if I include:
![image](https://github.com/AvaloniaUI/Avalonia/assets/18743473/ac5dd7f7-16ff-4b6e-b52d-b7e2146b4999)

Then it works correctly. Otherwise it always opens the home/user path.

**Reply by maxkatz6** [reply]
>     IStorageFolder? options = await storageProvider.TryGetWellKnownFolderAsync(WellKnownFolder.Downloads);

On linux TryGetWellKnownFolderAsync uses XDG_DOWNLOAD_DIR environment variable. If it's not defined, there is no folder.
BCL also doesn't support (yet) Environment.SpecialFolder.Downloads to make it work easier.

**Comment by drei97** [comment]
Is this working as intended (by design) or a Bug?

**Reply by antondie** [reply]
It seems to be related to #9383 -> "disable FreeDesktop file picker" where I had picked up the setting.

**Comment by maxkatz6** [comment, marked as answer]
FreeDesktop "open file dialog" doesn't support SuggestedStartLocation. Only save file dialog.
That's not a bug, but a limitation of the platform. GTK dialog might support it properly.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.FreeDesktop/DBusSystemDialog.cs#L52-L77

**Reply by drei97** [reply]
Thnx for explanation
